cmd/bucket-notification-handlers.go

```diff
@@ -170,6 +170,7 @@ func (api objectAPIHandlers) PutBucketNotificationHandler(w http.ResponseWriter,
 
 	rulesMap := config.ToRulesMap()
 	globalNotificationSys.AddRulesMap(bucketName, rulesMap)
+	globalNotificationSys.PutBucketNotification(ctx, bucketName, rulesMap)
```

Comment:
@Praveenrajmani , bd032d13ffe014b0a6f072ce87693a0df9037b86 moved all metadata to a single file in globalBucketMetadataSys and peer notification happens when the metadata is updated. Based on the bug you are seeing, looks like the globalNotificationSys rulesMap needs to be reloaded after peer-rest-server does a loadBucketMetadata.  Instead of bringing this old style back of each subsystem writing a handler to communicate with peers, would suggest fixing the notification system to reload when peer-server LoadBucketMetadataHandler is called.

Reply:
Yes correct @poornas 

The reason for doing that as a separate peer-rest call is because the loadBucketMetadata is one generic call that we do if any of the bucket metadata config gets updated. Like, any object locking config update, bucket policy update, quota etc..

Refreshing the rulesMap in peer's loadBucketMetadata func would be an overkill, As we will always refresh the rulesMap for any bucket metadata reloads which involves other updates like object lock, bucket policy, quota etc. And refreshing the rulesMap will also require loading bucket info from all the buckets.. This would be an overkill too..

Reply:
yes, that is true. Your approach is probably the best here..